### ISSUE TITLE
Panic upon unsupported provider id

### ISSUE THREAD

**Issue by ivan4th** [spacemeshos/go-spacemesh#4828]
## Description
go-spacemesh panics if the provider id is unsupported. For example, the following happens when provider id 0 is used on a machine without OpenCL. E.g. PoST data has been generated elsewhere, and then you try to run a node with that PoST data on a machine with no GPU:
```
2023-08-12T06:20:19.609Z        ERROR   24ce5.atxBuilder        Fatal: goroutine panicked. Stacktrace: goroutine 369312 [running]:
runtime/debug.Stack()
        /root/.go/src/runtime/debug/stack.go:24 +0x65
github.com/spacemeshos/go-spacemesh/log.Log.Panic({0xc0001e0f50?, {0x19fd97c?, 0xc0024bb1c0?}}, {0x1b59980, 0x19}, {0xc002c53dc0, 0x1, 0x1})
        /go-spacemesh/log/zap.go:46 +0xd7
github.com/spacemeshos/go-spacemesh/activation.(*Builder).StartSmeshing.func1()
        /go-spacemesh/activation/activation.go:222 +0x1e3
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /root/go/pkg/mod/golang.org/x/sync@v0.3.0/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
        /root/go/pkg/mod/golang.org/x/sync@v0.3.0/errgroup/errgroup.go:72 +0xa5
        {"node_id": "24ce5e6ee13ff06078a5fb3a10079dc9bc33dc5ac0706198ce346a4bd7e447c5", "module": "atxBuilder"}
2023-08-12T06:20:19.610Z        PANIC   24ce5.atxBuilder        initialization failed: invalid provider ID     {"node_id": "24ce5e6ee13ff06078a5fb3a10079dc9bc33dc5ac0706198ce346a4bd7e447c5", "module": "atxBuilder"}
```

It appears that some pre-validation of the provider id option wouldn't harm, as well as maybe some documentation updates regarding the possible values, e.g. `0` for GPU and `4294967295` for no GPU (or can -1 be specified as well?)

## Steps to reproduce

Run go-spacemesh with provider id `0` on a machine without GPU and OpenCL, e.g. when reusing instructions for running go-spacemesh found on the web

## Actual Behavior

Panic logged by go-spacemesh.
In the option description when running `go-spacemesh -h`, there is no explanation what `--smeshing-opts-provider` values are supported, other than -1 is the default:
```
      --smeshing-opts-provider int                     (default -1)
```

## Expected Behavior

go-spacemesh exiting with no panic but rather a proper error message regarding unsupported option value.
Some explanation of what values are supported in general and on the current system would be quite helpful, too

## Environment
Please complete the following information:
 - OS: Ubuntu 22.02 amd64
 - Node Version: v1.0.15
 - Smapp version (if applicable): no smapp

Panic location in the code:
https://github.com/spacemeshos/go-spacemesh/blob/fd9046c042e77a8dff53e6e8fa067d86d2ad420f/activation/activation.go#L222

**Comment by dshulyak** [comment]
@fasmat @poszu @pigmej someone knows if this still the case?

**Comment by fasmat** [comment]
This is resolved for v1.1.0 and later versions.

Provider ID is now optional and not evaluated if initialization is complete.

**Comment by dshulyak** [comment]
i will close it then
